src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1416,6 +1416,37 @@ private static bool CanResetFitViewToggle(object o)
             return true;
         }
 
+        /// <summary>
+        /// Selects an Element by ID and focuses the view around it
+        /// </summary>
+        /// <param name="id"></param>
+        private void FocusNode(object id)
+        {
+            try
+            {
+                var node = DynamoViewModel.Model.CurrentWorkspace.Nodes.First(x => x.GUID.ToString() == id.ToString());
+                if(node != null)
+                {
+                    //select the element
+                    DynamoSelection.Instance.ClearSelection();
+                    DynamoSelection.Instance.Selection.Add(node);
+
+                    DynamoViewModel.ShowElement(node, false);
+
+                    return;
+                }
+            }
+            catch
+            {
+                DynamoViewModel.Model.Logger.Log(Wpf.Properties.Resources.MessageFailedToFindNodeById);
+            }
+        }
+
+        private static bool CanFocusNode(object id)
+        {
+            return !string.IsNullOrEmpty(id.ToString());
```

Comment:
Shouldn't this return true or false depending on if the node is found for a valid id?

Reply:
Same as the above: 

This code is an almost identical duplication of the ``FindById`` and ``CanFindById`` methods that already existed. The modification was done to circumvent a limitation with Automatic mode and has been discussed with @QilongTang. 

![image](https://user-images.githubusercontent.com/5354594/173678575-b7401b72-a580-4b07-9dd6-4e14adbbea1e.png)

Reply:
I'm not sure what you mean by `The modification was done to circumvent a limitation with Automatic mode`. @QilongTang ?

Reply:
@aparajit-pratap These are used in view commands and the only available one we have now does not work in automatic run mode, it was set to exclude automatic mode specifically. So we decide to make a new command almost identical but enable node focus in automatic mode. The function does not have return type because it is how the function usually work for commands

Reply:
Ok, but why not log the same message (`MessageFailedToFindNodeById`) even if the node is not found (is null) and there is no exception thrown?